### PR TITLE
View list of incident reports

### PR DESCRIPTION
**Pull request by rileynwong**

### CODE REVIEW

**Comment by rileynwong** [comment]
r @kyle-rosenbluth ?

**Comment by sandlerben** [comment]
Can we close #10?

File: app/reports/views.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from flask.ext.login import login_required, current_user
+from . import reports
+from ..models import IncidentReport, Agency
+
+
+@reports.route('/all')
+@login_required
+def view_reports():
+    """View all idling incident reports."""
+    """Different roles have access to different reports.
```
Comment (sandlerben) [inline app/reports/views.py]: Nit: can we combine this into one docstring?

File: app/reports/views.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    A general user can see all reports submitted by that user."""
+
+    if current_user.is_admin():
+        reports = IncidentReport.query.all()
+        agencies = Agency.query.all()
+
+    elif current_user.is_agency_worker():
+        agency_id = current_user.agency.id
+        agencies = Agency.query.filter_by(id=agency_id)
+        reports = IncidentReport.query.filter_by(agency_id=agency_id)
```
Comment (sandlerben) [inline app/reports/views.py]: Couldn't we just do

`current_user.agency.reports`?

File: app/templates/reports/reports.html [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                </div>
+            </h2>
+
+            <div class="ui menu">
+                <div id="select-agency" class="ui dropdown item">
+                    <div class="text">
+                        {% if current_user.is_admin() %}
+                            All incident reports
+                        {% else %}
+                            {{ current_user.agency.name }}
```
Comment (sandlerben) [inline app/templates/reports/reports.html]: Maybe change to `{{ current_user.agency.name }}'s incident reports`

**Comment by sandlerben** [comment]
I'll review a bit more closely when this is all finished up (e.g. logic for general users). How do you think users should get to this page? Should we add an `reports` link to the navbar?

**Comment by rileynwong** [comment]
@sandlerben @thomaslee1911

File: app/reports/views.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    An Agency worker can all reports for their agency.
+    A general user can see all reports submitted by that user."""
+
+    if current_user.is_admin():
+        reports = IncidentReport.query.all()
+        agencies = Agency.query.all()
+
+    elif current_user.is_agency_worker():
+        reports = current_user.agency.reports
+        agencies = None
```
Comment (thomaslee1911) [inline app/reports/views.py]: Should the agency worker be associated with their agency here?

File: app/models/incident_report.py [hunk 5]
```diff
@@ -11,6 +11,9 @@ class Location(db.Model):
     incident_report_id = db.Column(db.Integer,
                                    db.ForeignKey('incident_reports.id'))
 
+    def __repr__(self):
+        return '<Location \'%s\'>' % self.longitude
```
Comment (thomaslee1911) [inline app/models/incident_report.py]: Not sure where this is used but maybe include latitude as well.

File: app/reports/__init__.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from flask import Blueprint
+
+reports = Blueprint('reports', __name__)
+
+from . import views  # noqa
```
Comment (thomaslee1911) [inline app/reports/__init__.py:5]: Could you do this in accordance with flake8 / without using noqa?

Comment (rileynwong) [inline app/reports/__init__.py:5]: no, i think this is what we do for any `__init__.py` file

File: app/models/incident_report.py (hunk 5 above)
Comment (rileynwong) [inline app/models/incident_report.py]: good call, i changed this to just return lat, long in string format

File: app/models/incident_report.py [hunk 7]
```diff
@@ -11,6 +11,9 @@ class Location(db.Model):
     incident_report_id = db.Column(db.Integer,
                                    db.ForeignKey('incident_reports.id'))
 
+    def __repr__(self):
+        return '%s, %s' % (self.latitude, self.longitude)
```
Comment (sandlerben) [inline app/models/incident_report.py]: Nit: Let's use string.format

http://stackoverflow.com/questions/5082452/python-string-formatting-vs-format

File: app/models/incident_report.py [hunk 8]
```diff
@@ -61,6 +65,7 @@ def flip_coin():
                 date=fake.date_time_between(start_date="-1y", end_date="now"),
                 duration=timedelta(minutes=randint(1, 30)),
                 agency=choice(agencies),
+                user=choice(users),
```
Comment (sandlerben) [inline app/models/incident_report.py:69]: So every report will have a user but it's possible not every user will have a report? Sounds good.

File: app/reports/__init__.py (hunk 6 above)
Comment (sandlerben) [inline app/reports/__init__.py:5]: she is right

File: app/reports/views.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
+from flask import render_template
+from flask.ext.login import login_required, current_user
+from . import reports
+from ..models import IncidentReport, Agency
+
+
+@reports.route('/all')
+@login_required
+def view_reports():
+    """View all idling incident reports.
```
Comment (sandlerben) [inline app/reports/views.py:10]: Nice docstring :+1:

File: app/reports/views.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+    if current_user.is_admin():
+        reports = IncidentReport.query.all()
+        agencies = Agency.query.all()
+
+    elif current_user.is_agency_worker():
+        reports = []
+        for agency in current_user.agencies:
+            reports.extend(agency.incident_reports)
+        agencies = None
```
Comment (sandlerben) [inline app/reports/views.py]: I would set agencies to None before the if/elif/else block. This takes advantage of what I consider to be a confusing and peculiar feature of python.

**Comment by sandlerben** [comment]
A few comments/questions on the page itself:
1. What does `Submitter` mean?
2. Can we format the date nicer?
3. Does it really make sense to have a location column?
4. There is a funky bug with the coloring of the switch.
5. When I select SEPTA and then All incident reports I see no reports in the table

**Comment by sandlerben** [comment]
@nancywong I'm seeing the warning "This pull request is closed, but the list-reports-b branch has unmerged commits."

Are all those unmerged commits on your new branch?

**Comment by rileynwong** [comment]
yes, everything's on #57, which was branched off of this w/ the new commits

**Pull request closed without merging**